proposals/improved-interpolated-strings.md

```diff
@@ -291,6 +291,39 @@ This was done to support partial formatting scenarios where the user wants to st
 introduce a bunch of unnecessary branches in standard interpolated string usage. We could consider an addendum where we use just `Format` methods if no `TryFormat` method is present, but
 it does present questions about what we do if there's a mix of both TryFormat and Format calls.
 
+### Passing previous arguments to the builder
+
+There is unfortunate lack of symmetry in the proposal at it currently exists: invoking an extension method in reduced form produces different semantics than invoking the extension method in
+normal form. This is different from most other locations in the language, where reduced form is just a sugar. We have a couple of potential options for resolving this:
+
+* Special case extension methods called in normal form. This feels pretty icky: why are extensions special here?
+* Allow other previous parameters to be passed to the builder. This gets complicated quickly: how do we determine what to pass to the builder? What if the builder has a `GetInterpolatedString`
+method that accepts the first parameter, but not the receiver, of an instance method?
+* Pass parameters to the builder marked with a specific attribute, a la `EnumeratorCancellation` support. This would need rules about whether we pass the receiver (maybe if the method is marked
+we pass the receiver, and we don't in the general case?), and what we do if parameters _after_ the string parameter are annotated, but it seems like a potential option.
+
+Some compromise is likely needed here, but either direction has complications. Some scenarios that would be affected by this is the `Utf8Formatter` below, or existing api patterns that have
+an `IFormatProvider` as the first argument.
```

Comment:
Thank you for adding this section in response to my pestering you about it. ;-)

I would like us to be able to add:
```C#
String.Format([InterpolatedStringArgument] IFormatProvider? provider, InterpolatedStringBuilder builder);
```
and have the provider passed to the GetInterpolatedStringBuilder method.

I would like us to be able to add:
```C#
Utf8Formatter.TryWrite([InterpolatedStringArgument] Span<byte> destination, Utf8InterpolatedStringBuilder);
```
and have the span passed to the builder as the destination.

This also provides a solution for using stack scratch space.  We can add an overload:
```C#
InterpolatedStringBuilder.GetInterpolatedStringBuilder(int baseLength, int formatHoleCount, Span<char> scratchSpace, out InterpolatedStringBuilder);
```
Then either we could expose a public API, or even if not internally use (and tell devs that want the interim solution to copy/paste it):
```C#
public static string Format([InterpolatedStringArgument] Span<char> scratchSpace, InterpolatedStringBuilder builder);
```
which we can then use like:
```C#
string result = Format(stackalloc char[256], $"{a} = {b}");
```